corda-api/build.gradle

```diff
@@ -21,13 +21,41 @@ dependencies {
         api project(':persistence')
         api project(':serialization')
 
-        api "javax.persistence:javax.persistence-api:$javaxPersistenceApiVersion"
-        api "net.corda.kotlin:kotlin-stdlib-jdk8-osgi:$kotlinVersion"
-        api "net.corda.kotlin:kotlin-stdlib-jdk7-osgi:$kotlinVersion"
-        api "org.osgi:osgi.annotation:$osgiVersion"
-        api "org.osgi:osgi.core:$osgiVersion"
-        api "org.osgi:org.osgi.service.component.annotations:$osgiScrAnnotationVersion"
-        api "org.slf4j:slf4j-api:$slf4jVersion"
+        api('javax.persistence:javax.persistence-api') {
+            version {
+                require javaxPersistenceApiVersion
```

Comment:
I am assuming without this instruction, Kotlin 1.7 also causes JPA version to upgrade?

Reply:
I am making the version constraints _explicit_, so that we know precisely how Gradle will interpret them.